MaxText/train.py

```diff
@@ -318,23 +324,31 @@ def train_loop(config, state=None):
   p_train_step = pjit(
     train_step,
     in_axis_resources=(state_mesh_annotations,
+                       None,
                        data_pspec,
                        None),
-    out_axis_resources=(state_mesh_annotations, None, None),
+    out_axis_resources=(state_mesh_annotations, None, None, None),
     static_argnums=(0,1,),
-    donate_argnums=2)
+    donate_argnums=(2,3,))
 
   example_batch = None
   last_step_completion = datetime.datetime.now()
+  grad_norms = deque(maxlen=config.grad_cut_window_len)
+  LARGE_GRADIENT_NORM = 1.0 * 10**6 # Used when gradient cutting is not desired.
 
   local_metrics_file = open(config.metrics_file, 'a', encoding="utf8") if config.metrics_file else None
 
   for step in np.arange(get_first_step(state), config.steps):
     example_batch = load_next_batch(train_iter, example_batch, config)
+    if step < config.grad_cut_start_step or not grad_norms:
```

Comment:
put this into its own function! and don't define `LARGE_GRADIENT_NORM`, I think `float('inf')` has the right semantics here.